### PR TITLE
IBFT2 roundchange not sent if timeout after import

### PR DESCRIPTION
**Pull request by rain-on**
There exists a window in which a "RoundTimeout" event can be inserted to the IBFT event queue, such that it is actioned _after_ a block has been successfully imported, but before the NewBlockEvent is actioned.

This results in an unrequired (and potentially incorrect) RoundChange message being sent to other validators. This change prevents this spurious transmission, which in turn reduces potential attack vectors.

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java [hunk 1]
```diff
@@ -140,7 +140,9 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
             id, privateTransaction, privateTransactionLookupId, Address.DEFAULT_PRIVACY);
       }
     } catch (final IllegalArgumentException | RLPException e) {
+      /// RIGHT HERE <----
```
Comment (usmansaleem) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java:143]: are these comments meant to be removed?

Comment (rain-on) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java:143]: oops

**Review by usmansaleem — APPROVED** [review]
LGTM

File: consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftController.java [hunk 2]
```diff
@@ -129,6 +130,14 @@ private void handleMessage(final Message message) {
   private <P extends IbftMessage<?>> void consumeMessage(
       final Message message, final P ibftMessage, final Consumer<P> handleMessage) {
     LOG.trace("Received IBFT {} message", ibftMessage.getClass().getSimpleName());
+
+    // Discard all messages which target the BLOCKCHAIN height (which SHOULD be 1 less than
+    // the currentHeightManager, but CAN be the same directly following import).
+    if (ibftMessage.getRoundIdentifier().getSequenceNumber()
+        <= blockchain.getChainHeadBlockNumber()) {
+      return;
```
Comment (jframe) [inline consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftController.java:138]: Think we should log if this happens

Comment (rain-on) [reply consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftController.java:138]: done

**Review by jframe — APPROVED** [review]

**Comment by saltiniroberto** [comment]
Fixes #1734

**Pull request merged**